### DISCUSSION TITLE
Reverse foreground and background on selection

### DISCUSSION THREAD

**Discussion by John-Gee** [wezterm/wezterm#3432] [General]
Hello,

When I select text I want to flip foreground and background color of the selected text. In Kitty setting selection_fg and selection_bg to none would do the trick but with wezterm it would just leave the colors as is.

Thank you!

**Comment by nekowinston** [comment]
An easy way would be to adjust your color scheme to [your own custom one](https://wezfurlong.org/wezterm/config/appearance.html#defining-a-color-scheme-in-your-weztermlua):
```lua
c = wezterm.config_builder()

local original = wezterm.color.get_builtin_schemes()["Catppuccin Mocha"]
local custom = wezterm.color.get_builtin_schemes()["Catppuccin Mocha"]
custom.selection_bg = original.selection_fg
custom.selection_fg = original.selection_bg
c.color_schemes = {
  ["Custom"] = custom,
}
c.color_scheme = "Custom"
```

**Reply by John-Gee** [reply]
That makes sense. Wouldn't it be simpler to just use a temp var to switch selection_bg and selection_fg of c itself?

Thanks a lot!

edit: nope that did not work because I'm assuming color.selection_fg/bg is not defined without using those schemes...

**Reply by JordanP-Dev** [reply]
You mean you want to switch the fg / bg colors in the selected text, like this (from xfce-terminal):

![reverse_color_selection](https://user-images.githubusercontent.com/29902477/229546181-54773b51-99a8-4355-a2f4-bbaead49c598.png)

There is an option to do this for the cursor: https://wezfurlong.org/wezterm/config/lua/config/force_reverse_video_cursor.html, but I don't think there is an option for selections.

**Reply by John-Gee** [reply]
Yup exactly like in your picture!
It's more of a habit from my previous terminal than a need, but I kind of want to start with the same look for now.
Thank you!

**Reply by John-Gee** [reply]
I've opened #3440 for this.
